File: easy/1304_find_n_unique_integers_sum_up_to_zero/Solution.java
class Solution {
  public int[] sumZero(int n) {
    final int[] nums = new int[n];

    int i = 0;

    if (n % 2 == 1) {
      nums[i] = 0;
      ++i;
    }

    for (int num = 1; num < n / 2 + 1; ++num) {
      nums[i] = num;
      nums[i + 1] = -num;
      i += 2;
    }

    return nums;
  }
}

File: easy/1304_find_n_unique_integers_sum_up_to_zero/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<int> sumZero(int n) {
    std::vector<int> nums;
    nums.reserve(n);

    if (n % 2 == 1) {
      nums.push_back(0);
      --n;
    }

    for (int num = 1; num < n / 2 + 1; ++num) {
      nums.push_back(num);
      nums.push_back(-num);
    }

    return nums;
  }
};
